core-bundle/src/Fragment/Annotation/FrontendModule.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Fragment\Annotations;
+
+use Doctrine\Common\Annotations\Annotation\Attribute;
+use Doctrine\Common\Annotations\Annotation\Attributes;
+use Doctrine\Common\Annotations\Annotation\Target;
+
+/**
+ * Annotation that can be used to define controller as a frontend module.
+ *
+ * @Annotation
+ * @Target("CLASS", "METHOD")
+ * @Attributes({
+ *     @Attribute("category", required = true, type = "string"),
+ *     @Attribute("options", type = "array"),
+ *     @Attribute("renderer", type = "string"),
+ *     @Attribute("service", type = "string"),
+ *     @Attribute("template", type = "string"),
+ *     @Attribute("type", type = "string"),
+ * })
+ */
+final class FrontendModule
```

Comment:
Can‘t we use an abstract fragment parent class? They look identical :)

Reply:
Just did that in 12344d203, although not sure if it's a good idea since we define the annotation details in child class … @aschempp ?

Reply:
I'm fine with that, because it reduces duplicate code.